app/models/comment.rb

```diff
@@ -0,0 +1,13 @@
+class Comment < ActiveRecord::Base
+
+  has_many :comments, as: :commentable
+
+  belongs_to :commentable, polymorphic: true
+  belongs_to :user
+
+  validates :content, presence: true
+  validates :user, presence: true
```

Comment:
if you like the compactness, you can combine presence validations like so:
```
validates :content, :user, presence: true
```

Also, do you want to be validating the presence of `commentable` as well?